routers/install/install.go

```diff
@@ -148,8 +148,8 @@ func Install(ctx *context.Context) {
 
 	// Server and other services settings
 	form.OfflineMode = setting.OfflineMode
-	form.DisableGravatar = false       // when installing, there is no database connection so that given a default value
-	form.EnableFederatedAvatar = false // when installing, there is no database connection so that given a default value
+	form.DisableGravatar = false                      // when installing, there is no database connection so that given a default value
+	form.EnableFederatedAvatar = !setting.InstallLock // when installing, there is no database connection so that given a default value
```

Comment:
Actually, could it make sense to completely DROP the setting frmo the install screen ? Isn't it confusing to set in-db thing from there ?

Reply:
imo since it has been configurable in the past we should try to keep it here.